lib/scanner.js

```diff
@@ -154,12 +154,33 @@ function scanQR (callback) {
   })
 }
 
-function scanPDF417 (callback) {
+function scanPDF417 (callback, idCardStillsCallback) {
+  let captureIdCardPhoto = false
+  let idCardTimer = 4000
+  let endInterval = 24000
+  let handleIdCardPhoto = setInterval(() => {
+    captureIdCardPhoto = true
+    idCardTimer = idCardTimer + 4000
+  }, idCardTimer)
```

Comment:
Incrementing the timer like this would make the interval increasingly longer. Your `idCardTimer` has the following values on each iteration [4000, 8000, 12000, 16000, 20000]. This would work if you were working with `setTimeout()`, but on intervals, you only need to pass the timer for each iteration, so no need to increment it.
I would do something slightly different, something of this sort:

```
const ID_CARD_CAPTURE_INCREMENT = 4000
const PHOTO_LIMIT = 5
let photosTaken = 0
const handleIdCardPhoto = setInterval(() => {
  captureIdCardPhoto = true
  photosTaken++
}, ID_CARD_CAPTURE_INCREMENT)

...

if (photosTaken >= PHOTO_LIMIT) clearInterval(handleIdCardPhoto)
```

Reply:
Actually, I think the interval stays the same because it uses the initial timer value when setInterval() is called. But it's way cleaner what you suggested. Thanks!